Duplicati/Library/Snapshots/WindowsSnapshot.cs

```diff
@@ -108,59 +105,94 @@ public WindowsSnapshot(string[] sourcepaths, Dictionary<string, string> options)
+                                                if (Directory.Exists(file.Path + "\\"))
+                                                {
+                                                    m_sourcepaths.Add(file.Path + "\\");
+                                                    Logging.Log.WriteMessage(string.Format("For VM {0} - adding {1}.", component.ComponentName, file.Path + "\\"), Logging.LogMessageType.Profiling);
+                                                }
+                                            }
+                                            else
+                                            {
+                                                if (File.Exists(Path.Combine(file.Path, file.FileSpecification)))
+                                                {
+                                                    m_sourcepaths.Add(Path.Combine(file.Path, file.FileSpecification));
+                                                    Logging.Log.WriteMessage(string.Format("For VM {0} - adding {1}.", component.ComponentName, Path.Combine(file.Path, file.FileSpecification)), Logging.LogMessageType.Profiling);
+                                                }
+                                            }
+                            }
+                            else
+                                throw new Exception("Microsoft Hyper-V VSS Writer not found - cannot backup Hyper-V machines.");
+                        }
+                        finally
+                        {
+                            m_backup.FreeWriterMetadata();
+                        }
+                    }
+                    else
+                    {
+                        Logging.Log.WriteMessage("This is client version of Windows. Hyper-V VSS writer is present only on Server version. Backup will continue, but will be crash consistent only in opposite to application consistent in Server version.", Logging.LogMessageType.Warning);
+
+                        foreach (var hyperVGuest in hyperVGuests)
+                            if (requestedHyperVMs.Contains(hyperVGuest.ID))
+                                foreach (var path in hyperVGuest.DataPaths)
                                 {
-                                    if (file.FileSpecification.Contains("*")) 
-                                        continue;
-                                    m_sourcepaths.Add(file.Path + file.FileSpecification);
+                                    m_sourcepaths.Add(path);
+                                    Logging.Log.WriteMessage(string.Format("For VM {0} - adding {1}.", hyperVGuest.ID, path), Logging.LogMessageType.Profiling);
                                 }
-                            }
                     }
 
-                    m_backup.FreeWriterMetadata();
+                    m_sourcepaths = m_sourcepaths.Distinct(StringComparer.CurrentCultureIgnoreCase).OrderBy(a => a).ToList();
                 }

```

Comment:
This should use `Library.Utility.Utility.ClientFilenameStringComparer` to be consistent, but since Hyper-V is Windows-only it is not terribly important